analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContext.cs

```diff
@@ -0,0 +1,169 @@
+        {
+            throw new NotImplementedException("Add a reference to the Microsoft.CodeAnalysis.VisualBasic.Workspaces package.");
+        }
+        else
+        {
+            throw new ArgumentException("Invalid type parameter.", nameof(TLanguageKindEnum));
+        }
+    }
+
+    public void RegisterOperationAction(Action<OperationAnalysisContext> action, ImmutableArray<OperationKind> operationKinds) =>
+        registerOperationAction(RoslynSymbolStartAnalysisContext, action, operationKinds);
+
+    public void RegisterOperationBlockAction(Action<OperationBlockAnalysisContext> action) =>
+        registerOperationBlockAction(RoslynSymbolStartAnalysisContext, action);
+
+    public void RegisterOperationBlockStartAction(Action<OperationBlockStartAnalysisContext> action) =>
+        registerOperationBlockStartAction(RoslynSymbolStartAnalysisContext, action);
+
+    public void RegisterSymbolEndAction(Action<SymbolAnalysisContext> action) =>
+        registerSymbolEndAction(RoslynSymbolStartAnalysisContext, action);
+
+    public void RegisterSyntaxNodeAction<TLanguageKindEnum>(Action<SyntaxNodeAnalysisContext> action, params TLanguageKindEnum[] syntaxKinds) where TLanguageKindEnum : struct
+    {
+        var languageKindType = typeof(TLanguageKindEnum);
+        if (languageKindType == typeof(CS.SyntaxKind))
+        {
+            registerSyntaxNodeActionCS(RoslynSymbolStartAnalysisContext, action, syntaxKinds.Cast<CS.SyntaxKind>().ToImmutableArray());
+        }
+        else if (languageKindType.FullName == "Microsoft.CodeAnalysis.VisualBasic.SyntaxKind")
+        {
+            throw new NotImplementedException("Add a reference to the Microsoft.CodeAnalysis.VisualBasic.Workspaces package.");
+        }
+        else
+        {
+            throw new ArgumentException("Invalid type parameter.", nameof(TLanguageKindEnum));
+        }
+    }
+
+    // symbolStartAnalysisContextParameter => ((symbolStartAnalysisContextType)symbolStartAnalysisContextParameter)."propertyName"
+    private static Func<object, TProperty> CreatePropertyAccessor<TProperty>(Type symbolStartAnalysisContextType, string propertyName)
+    {
+        var symbolStartAnalysisContextParameter = Parameter(typeof(object));
+        return Lambda<Func<object, TProperty>>(
+            Property(
+                Convert(symbolStartAnalysisContextParameter, symbolStartAnalysisContextType), propertyName),
+            symbolStartAnalysisContextParameter).Compile();
+    }
+
+    // (object symbolStartAnalysisContextParameter, Action<TContext> registerActionParameter)
+    //     => ((symbolStartAnalysisContextType)symbolStartAnalysisContextParameter)."registrationMethodName"<typeArguments>(registerActionParameter)
+    private static Action<object, Action<TContext>> CreateRegistrationMethod<TContext>(Type symbolStartAnalysisContextType, string registrationMethodName, params Type[] typeArguments)
+    {
+        var receiver = Parameter(typeof(object));
```

Comment:
Personal and very very subjective opinion: `receiver` is very message-oriented. `target` is more method-oriented, IMO.

Reply:
We are a bit inconsistent here. We use "instance" and "sender" in the shim layer e.g., here:
https://github.com/SonarSource/sonar-dotnet/blob/a195fac7f6c8cf7f5edce3b0da1b3b28bd8edbf2/analyzers/src/SonarAnalyzer.CFG/ShimLayer/LightupHelpers.cs#L547-L550

I used "Receiver" because of https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.imethodsymbol.receivertype?view=roslyn-dotnet-4.7.0

"Target" isn't used, as far as I know. "ThisParameter" is another option that is quite common. I would use "Sender" or "Instance" to align with LightupHelpers jargon.

Reply:
Ok, I see now why you picked `receiver`. Good for me.